programs/survival/renewable_elytra_trade.sc

```diff
@@ -0,0 +1,45 @@
+// Customize the trade
+
+global_trade = encode_nbt(
+	{
+		'sell' -> {'id' -> 'minecraft:elytra', 'Count'->1},
+		'buy' -> {'id' -> 'minecraft:emerald', 'Count' -> 40},
+		'buyB' -> {'id' -> 'minecraft:phantom_membrane', 'Count' -> 5},
+		'uses' -> 0,
+		'demand' -> -20,
+		'priceMutiplier' -> 0.0,
+		'specialPrice' -> 0,
+		'rewardExp' -> 1, // boolean (gives/doesn't give xp)
+		'xp' -> 1, // count
+	}
+);
+
+//set the changes of the trade appearing in %
+global_chances = 100;
+
+// Customize the conditions
+_add_trade(e, p) -> (
+	nbt = e~'nbt';
+	if(
+		//e~'dimension'=='the_end' && //villager needs to be in the end
+		nbt:'VillagerData':'level'>0 && // minimum level at which the trade appears
+		he = (p~'holds':0 == 'elytra') && //does the player need to hold an elytra?
+		parse_nbt(nbt:'Tags')~'HasTrade' == null,
```

Comment:
`HasTrade` is a really generic tag
I suggest `renewable_elytra_trade.sc-HasTrade` or something similar

Reply:
I also suggest putting this check at the front, so it fails immediately for villagers which already have it, thus not having to  do redundant checks/calculations